### ISSUE TITLE
Make mock-debug a yeoman generator, publish /common as NPM module

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/vscode#1356]
The whole process of cloning mock-debug to make a debugger extension and then changing it is cumbersome. Like for normal extensions, there should be a yeoman generator that asks you questions about what languages the debugger is for, what the name should be etc.

Secondly, in the docs you express the intent to move the common part of the mock-debug repo to a NPM module, which makes sense, as it's the TS implementation of the debugger protocol so we can write extensions in TS using it. You would be able to add more features over time etc.

I think this would make writing debugger extensions more straight forward.

**Comment by weinand** [comment]
@felixfbecker a first cut of the npm modules is available as vscode-debugprotocol and vscode-debugadapter. The corresponding github repo will be made pulic soon.

**Comment by felixfbecker** [comment]
@weinand   a little weird having an npm module just for the protocol .d.ts. Maybe this should rather be put to DefiniatelyTyped? And is a yeoman generator planned?

**Comment by weinand** [comment]
@felixfbecker as I said, this is a first cut. We might want to add more to the vscode-debugprotocol module.

We don't want to use DefinitelyTyped for this because we need the strong version support of npm.

Yes, a yeoman generator for an empty debugadapter is planned but first we had to refactor the code because we don't want to have the generator generate lots of duplicated code (and I'm a bit reluctant with the yeoman generator in general because node based debug-adapters implemented in TypeScript are not as frequently created as VSCode extensions in general).

BTW, I just made Microsoft/vscode-debugadapter-node public.
